### ISSUE TITLE
ebay-icon: aria-labelledby should not be applied when a11y-text is not provided

### ISSUE THREAD

**Issue by melquan** [eBay/ebayui-core#1162]
# Bug Report

## Description
`aria-labelledby` is being applied even when `a11y-text` is not provided:

https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/index.marko#L21

Proposed solution:
```
aria-labelledby:scoped=(input.a11yText && 'text')
```

## Workaround
None

**Comment by ianmcburnie** [comment]
@leksterz Another possible one for you to look into

**Comment by msendlakowski** [comment]
@ianmcburnie  - this is more critical with a recent update to the `@testing-library`.   It throws a hard to debug error if an aria-labelledby doesn't have a valid target.  So unit tests will fail with the following error:

```
TypeError: Cannot read property 'getAttribute' of null

      39 | 
      40 |         test('then the overlay will render', async() => {
    > 41 |             expect(getByLabelText("sample a11y text")).toBeTruthy();
         |                    ^
      42 |         });

```

**Comment by ianmcburnie** [comment]
Thanks @msendlakowski what version do you need the patch in?

**Comment by msendlakowski** [comment]
Yeah, I'm testing with 5.6.0 now, so that would be great.

**Comment by ianmcburnie** [comment]
@DylanPiercey I don't understand why the `aria-labelledby` attribute is being rendered, when the scoped id it depends upon is behind a failing conditional test:

https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/index.marko#L56

**Comment by DylanPiercey** [comment]
@ianmcburnie the issue is not on the line that you linked, but instead https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/index.marko#L23

It unconditionally is adding the `aria-labelledby:scoped="text"`. As @melquan mentioned it should be `aria-labelledby:scoped=(input.a11yText && 'text')` to only add the `aria-labelledby` if there was `a11yText`.

**Comment by ianmcburnie** [comment]
Actual fix required, for those following along, was:

`aria-labelledby=(input.a11yText && component.elId("text"))>`

Thanks for the help, @DylanPiercey !

**Comment by ianmcburnie** [comment]
Fixed and published in v10.6.1
